File: android/app/src/main/java/com/rork/porchivo/ui/screens/SafetyScreen.kt
package com.rork.porchivo.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.TrendingDown
import androidx.compose.material.icons.filled.TrendingUp
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.navigation.NavController
import com.rork.porchivo.config.AppConfig
import com.rork.porchivo.ui.components.NeedleGauge
import com.rork.porchivo.ui.theme.PorchivoTheme
import com.rork.porchivo.ui.viewmodel.ShipmentsViewModel
import com.rork.porchivo.util.RiskEngine
import com.rork.porchivo.util.SafetyScore

/** Porch risk breakdown — mirrors the Expo app's safety-score / porch-risk screens. */
@Composable
fun SafetyScreen(
    navController: NavController,
    modifier: Modifier = Modifier,
    shipmentsViewModel: ShipmentsViewModel = viewModel(),
) {
    val c = PorchivoTheme.colors
    val myShipments by shipmentsViewModel.myShipments.collectAsStateWithLifecycle()
    val factors = remember(myShipments) { RiskEngine.factors(myShipments) }
    val riskScore = remember(myShipments) { RiskEngine.score(myShipments) }
    // Displayed as a SAFETY score (higher = safer) via the shared helper.
    val safetyScore = remember(riskScore) { SafetyScore.fromRisk(riskScore) }
    val band = SafetyScore.band(safetyScore)
    val tint = when (band) {
        RiskEngine.RiskLevel.HIGH -> c.danger
        RiskEngine.RiskLevel.MEDIUM -> c.warmOrange
        RiskEngine.RiskLevel.LOW -> c.success
    }

    Column(
        modifier = modifier
            .fillMaxSize()
            .background(c.background)
            .verticalScroll(rememberScrollState()),
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 4.dp, vertical = 4.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            IconButton(onClick = { navController.popBackStack() }) {
                Icon(
                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                    contentDescription = "Back",
                    tint = c.textPrimary,
                )
            }
            Text(
                text = "Safety Score",
                color = c.textPrimary,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
            )
        }

        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(14.dp),
        ) {
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = c.surface),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
            ) {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(20.dp),
                    horizontalAlignment = Alignment.CenterHorizontally,
                ) {
                    NeedleGauge(
                        score = safetyScore,
                        riskLabel = band.label,
                        scoreColor = tint,
                        modifier = Modifier.fillMaxWidth(),
                    )
                    Spacer(modifier = Modifier.height(12.dp))
                    Text(
                        text = "Higher is safer — protections add points, risks subtract them.",
                        color = c.textSecondary,
                        fontSize = 12.sp,
                        lineHeight = 17.sp,
                    )
                    Spacer(modifier = Modifier.height(10.dp))
                    Text(
                        text = "${AppConfig.SocialProof.PACKAGES_STOLEN_STAT} packages were stolen in the US last year, and " +
                            "${AppConfig.SocialProof.STOLEN_RATIO} Americans have had a package stolen (SafeWise · Security.org, 2025).",
                        color = c.textSecondary,
                        fontSize = 12.sp,
                        lineHeight = 17.sp,
                    )
                }
            }

            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = c.surface),
                elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text(
                        text = "SAFETY FACTORS",
                        color = c.textMuted,
                        fontSize = 11.sp,
                        fontWeight = FontWeight.SemiBold,
                        letterSpacing = 1.4.sp,
                    )
                    Spacer(modifier = Modifier.height(8.dp))
                    factors.forEachIndexed { index, factor ->
                        if (index > 0) HorizontalDivider(color = c.border)
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(vertical = 12.dp),
                            verticalAlignment = Alignment.CenterVertically,
                            horizontalArrangement = Arrangement.spacedBy(10.dp),
                        ) {
                            val safetyDelta = -factor.delta
                            Icon(
                                imageVector = if (safetyDelta > 0) Icons.Filled.TrendingUp else Icons.Filled.TrendingDown,
                                contentDescription = null,
                                tint = if (safetyDelta > 0) c.success else c.danger,
                                modifier = Modifier.size(16.dp),
                            )
                            Text(
                                text = factor.label,
                                color = c.textPrimary,
                                fontSize = 14.sp,
                                fontWeight = FontWeight.Medium,
                                modifier = Modifier.weight(1f),
                            )
                            Text(
                                text = if (safetyDelta > 0) "+$safetyDelta" else "$safetyDelta",
                                color = if (safetyDelta > 0) c.success else c.warmOrange,
                                fontSize = 14.sp,
                                fontWeight = FontWeight.Bold,
                            )
                        }
                    }
                }
            }

            Spacer(modifier = Modifier.height(12.dp))
        }
    }
}

File: ios/Porchivo/Views/SafetyScreen.swift
//
//  SafetyScreen.swift
//  Porchivo
//
//  Porch risk breakdown — gauge, contributing factors, theft stats, tips.
//

import SwiftUI

struct SafetyScreen: View {
    @Environment(AppState.self) private var appState
    @Environment(\.porchivo) private var c

    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                gaugeCard
                factorsCard
                statsCard
                tipsCard
            }
            .padding(.horizontal, 16)
            .padding(.top, 16)
            .padding(.bottom, 120)
        }
        .background(c.background.ignoresSafeArea())
        .navigationTitle("Safety Score")
        .navigationBarTitleDisplayMode(.inline)
    }

    private var gaugeCard: some View {
        let safety = SafetyScore.value(fromRisk: RiskEngine.score(appState.shipments))
        let band = SafetyScore.band(safety)
        let tint: Color = band == .high ? c.danger : (band == .medium ? c.warmOrange : c.success)
        return Card {
            VStack(spacing: 12) {
                NeedleGaugeView(score: safety, riskLabel: band.label, scoreColor: tint)
                Text("Higher is safer — protections add points, risks subtract them.")
                    .font(.system(size: 12))
                    .foregroundStyle(c.textSecondary)
                    .multilineTextAlignment(.center)
            }
            .frame(maxWidth: .infinity)
        }
    }

    private var factorsCard: some View {
        let factors = RiskEngine.factors(appState.shipments)
        return Card {
            VStack(alignment: .leading, spacing: 12) {
                Text("Contributing factors")
                    .font(.system(size: 16, weight: .bold))
                    .foregroundStyle(c.textPrimary)
                ForEach(factors) { f in
                    let safetyDelta = -f.delta
                    HStack {
                        Image(systemName: safetyDelta > 0 ? "checkmark.circle.fill" : "exclamationmark.triangle.fill")
                            .foregroundStyle(safetyDelta > 0 ? c.success : c.warmOrange)
                        Text(f.label)
                            .font(.system(size: 13, weight: .medium))
                            .foregroundStyle(c.textPrimary)
                        Spacer()
                        Text(safetyDelta > 0 ? "+\(safetyDelta)" : "\(safetyDelta)")
                            .font(.system(size: 13, weight: .bold))
                            .foregroundStyle(safetyDelta > 0 ? c.success : c.warmOrange)
                    }
                }
            }
        }
    }

    private var statsCard: some View {
        Card {
            VStack(alignment: .leading, spacing: 10) {
                Text("Package theft in the U.S.")
                    .font(.system(size: 16, weight: .bold))
                    .foregroundStyle(c.textPrimary)
                statRow("\(AppConfig.SocialProof.packagesStolenStat)", "packages stolen in the US last year — SafeWise 2025 report")
                statRow("\(AppConfig.SocialProof.stolenRatio)", "Americans have had a package stolen — Security.org 2025 survey")
                statRow("$15B", "lost to porch piracy last year — SafeWise 2025 report")
            }
        }
    }

    private func statRow(_ value: String, _ label: String) -> some View {
        HStack(spacing: 12) {
            Text(value)
                .font(.system(size: 22, weight: .black))
                .foregroundStyle(c.accent)
                .frame(width: 70, alignment: .leading)
            Text(label)
                .font(.system(size: 12))
                .foregroundStyle(c.textSecondary)
        }
    }

    private var tipsCard: some View {
        Card {
            VStack(alignment: .leading, spacing: 10) {
                Text("Raise your score")
                    .font(.system(size: 16, weight: .bold))
                    .foregroundStyle(c.textPrimary)
                tip("Assign a Porch Partner for active deliveries", "person.2.fill", c.success)
                tip("Add drop-off instructions for couriers", "text.bubble.fill", c.accent)
                tip("Keep deliveries in daytime windows when possible", "sun.max.fill", c.warmOrange)
                tip("Install a doorbell camera for visible deterrence", "video.fill", c.danger)
            }
        }
    }

    private func tip(_ text: String, _ symbol: String, _ tint: Color) -> some View {
        HStack(spacing: 10) {
            Image(systemName: symbol).foregroundStyle(tint)
            Text(text).font(.system(size: 13, weight: .medium)).foregroundStyle(c.textPrimary)
            Spacer()
        }
    }
}
